src/evm/protocol/erc20.rs

```diff
@@ -0,0 +1,38 @@
+use crate::protocol::errors::SimulationError;
+use alloy_primitives::{bytes::Bytes, Address};
+
+/// Safely converts a `Bytes` object to an `Address` object.
+///
+/// Checks the length of the `Bytes` before attempting to convert, and returns a `SimulationError`
+/// if not 20 bytes long.
+pub fn bytes_to_erc20_address(address: &Bytes) -> Result<Address, SimulationError> {
+    if address.len() == 20 {
+        Ok(Address::from_slice(address))
+    } else {
+        Err(SimulationError::InvalidInput(
+            format!("Invalid ERC20 token address: {:?}", address),
+            None,
+        ))
+    }
+}
```

Comment:
I think we should consider making a trait for this in `tycho-ethereum` (like we did for ethers type [here](https://github.com/propeller-heads/tycho-indexer/blob/82f9974a864a5d4ae7f216e64b74c5f73c020229/tycho-ethereum/src/lib.rs#L51-L127)) so we can reuse it, for example in the token-quoter repo

Reply:
Nice idea - not urgent for the release though I guess?